Pixi bindings

whitetigle opened the pull request:
Hi, 
here are the bindings for **pixi**, **pixi-sound** and **pixi-particles**.
`dotnet restore `and `dotnet pack` seem to work on my machine.

I'm not sure everything's all right though since thats the very first packages I push to this repo. So thanks in advance for reviewing this PR!

MangelMaxime commented on Pixi/Fable.Import.Pixi.fsproj:
```diff
@@ -0,0 +1,12 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <Version>0.1.0</Version>
```
You need to set this version to 0.0.0.

The version will be updated by the `build.fsx` according to the RELEASE_NOTES.md version.

MangelMaxime commented:
You also need to add the folder path to [this list](https://github.com/fable-compiler/fable-import/blob/master/build.fsx#L13)

MangelMaxime commented:
@alfonsogarciacaro 
I think all the import like : `[<Import("sound.SoundSprite","PIXI")>]` should be `[<Import("SoundSprite","PIXI/sound")>]` but perhaps I am wrong.

In general, in JS the import are of the form: `import { exportedThing } from "my/Path"`

whitetigle commented:
ok I'll make the changes later when I come back. Thanks @MangelMaxime  👍

MangelMaxime commented on Pixi/README.md:
```diff
@@ -0,0 +1,15 @@
+
+### Using
+1. `paket add Fable.Import.Pixi`
+2. `yarn add pixi.js pixi-particles pixi-sound`
```
@whitetigle 

Didn't you say that all the libs was coming from the same module ?
Here you are adding one module per import. So, here you could use pixi without pixi-sound and pixi-particle.

So, it's should be 3 separate nuget packages.

whitetigle replied:
No that's not what I meant. But I may have misunderstood your point though 😉 

So,
Pixi does neither need pixi-sound nor pixi-particles, 
But Pixi-sound needs pixi and pixi-particles too.

So what am I to do?

MangelMaxime replied:
Ah ok :)

No problem, I didn't understood the right thing them :)

So you need to create 3 packages:
- Fable.Import.Pixi
- Fable.Import.PixiSound (or Fable.Import.Pixi.Sound)
- Fable.Import.PixiParticle (or Fable.Import.Pixi.Particle)

Then you can use, `ProjectReference` in the fsproj of PixiSound and PixiParticle to make then dependant of Pixi.
As an example: https://github.com/fable-compiler/fable-react/blob/master/src/Fable.ReactLeaflet/Fable.ReactLeaflet.fsproj#L10-L12

Then, in the `build.fsx` make sure to list the project in the correct order:

1. First Pixi
2. Then PixiSound and PixiParticle

This will ensure, that if we update Pixi and PixiSound for example then we release Pixi first and update it's version in the fsproj and so PixiSound will reference the right version of Pixi too.

whitetigle replied:
ok I'll do that tomorrow. Thanks @MangelMaxime 👍

alfonsogarciacaro commented:
I agree with @MangelMaxime comments 😸 

I also think `[<Import("sound.SoundSprite","PIXI")>]` should be changed to  `[<Import("SoundSprite","PIXI/sound")>]` but tbh I don't remember exactly how it works 😅 I guess we have to try to see if it works with the slashes. If it does, maybe the dot in "PIXI.ticker" and "PIXI.utils" needs to be replaced by a slash too.

whitetigle commented:
Ok I made the updates.

We now have 4 repo:
- Pixi
- Pixisound
- PixiParticles
- Animejs (I added this since it works)

I have checked that all repo pack well and also that that Fable-Import Build target works. 
I have changed the import directives too and checked the updated bindings work in samples-pixi (an update is also planned for this repo).

So we should be all good now. Can you check please?

MangelMaxime commented on PixiParticles/paket.references:
```diff
@@ -0,0 +1,3 @@
+FSharp.Core
+Fable.Core
+Fable.Import.Browser
```
I think PixiParticles don't need a deps over Fable.import.Browser as it doesn't use any type coming from this namespace.

I just read the code didn't check with VSCode so can be wrong

whitetigle replied:
ok should be good now :)

MangelMaxime commented:
Thanks @whitetigle 

Do you have a nuget.org account ? So I can make you owner of the Pixi too.

MangelMaxime commented:
@whitetigle  @alfonsogarciacaro I am taking the action to create the package on nuget.org

whitetigle commented:
@MangelMaxime nuget nickname: whitetigle

whitetigle commented:
Thanks for your patience Maxime! 🥇

MangelMaxime commented:
@alfonsogarciacaro @whitetigle The packages are now published and you should have 4 invitations waiting.

MangelMaxime commented:
@whitetigle @alfonsogarciacaro 

Ok, seems like: `[<Import("SoundSprite","PIXI/sound")>]` was a bad idea.

`[<Import("SoundSprite","PIXI.sound")>]` is the right way to do it (at least it's working). I tested it on my version of **samples-pixi**.

whitetigle commented:
ok! 👍 Thanks @MangelMaxime
